toolchain/parser/parser_handle_array_expression.cpp

```diff
@@ -0,0 +1,51 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lexer/token_kind.h"
+#include "toolchain/lexer/tokenized_buffer.h"
+#include "toolchain/parser/parse_node_kind.h"
+#include "toolchain/parser/parser_context.h"
+#include "toolchain/parser/parser_state.h"
+
+namespace Carbon {
+
+auto ParserHandleArrayExpression(ParserContext& context) -> void {
+  auto state = context.PopState();
+  state.state = ParserState::ArrayExpressionSemi;
+  context.AddNode(ParseNodeKind::ArrayExpressionStart,
+                  context.ConsumeChecked(TokenKind::OpenSquareBracket),
+                  state.subtree_start, state.has_error);
```

Comment:
```suggestion
  context.AddLeafNode(ParseNodeKind::ArrayExpressionStart,
                      context.ConsumeChecked(TokenKind::OpenSquareBracket),
                      state.has_error);
```
There's a slightly simpler function you can use for parse tree nodes like `ArrayExpressionStart` that don't have children.